src/resolve/resolved-dns-stream.c

```diff
@@ -341,6 +374,9 @@ DnsStream *dns_stream_unref(DnsStream *s) {
                 s->manager->n_dns_streams--;
         }
 
+        if (s->tfo_address)
+                mfree(s->tfo_address);
```

Comment:
no mfree() here, as you aren't interested in the return value. Moreover, the if check is redundant, both glibc free() and our mfree() wrapper are fine with a NULL parameter, and become NOPs in that case